### DISCUSSION TITLE
What all hardware features are supported in zephyr for STM32U5 series MCU ?

### DISCUSSION THREAD

**Discussion by enthunilu** [zephyrproject-rtos/zephyr#48028] [General] [answered]
Hi,

I am exploring STM32U5 series MCU's for system low power requirement.
I could find some links for zephyr feature support on STM32U575 
[https://docs.zephyrproject.org/latest/boards/arm/nucleo_u575zi_q/doc/nucleou575zi_q.html](https://docs.zephyrproject.org/latest/boards/arm/nucleo_u575zi_q/doc/nucleou575zi_q.html)
![image](https://user-images.githubusercontent.com/93653544/179798417-ce3e4b5a-c46c-4e67-a72e-239edc2941f7.png)

and STM32U585
[https://docs.zephyrproject.org/latest/boards/arm/b_u585i_iot02a/doc/index.html](https://docs.zephyrproject.org/latest/boards/arm/b_u585i_iot02a/doc/index.html)
![image](https://user-images.githubusercontent.com/93653544/179798323-2f70f05e-14e8-417a-bee8-b79ac1cbbcb1.png)

I wanted to know is there any other link or list of features/hardware peripheral support that are planned for STM32U5 series controllers?

Also, does STM32U5 has MCUboot support ?

Can anyone please help ?

Thanks!!

**Comment by erwango** [comment, marked as answer]
Hi,

For supporeted HW, I suggest to look at https://github.com/zephyrproject-rtos/zephyr/tree/main/dts/arm/st/u5 and see available nodes.
Regarding the not yet supported features, here are the on going developments:
https://github.com/zephyrproject-rtos/zephyr/pulls?q=is%3Aopen+label%3A%22platform%3A+STM32%22+STM32U5+is%3Apr+-label%3ARFC
I'm not aware of more plans beyond this list for now.

**Reply by enthunilu** [reply]
Thanks @erwango for the update. 
One more question, does we have support of MCUboot for STM32U5 ?

**Reply by enthunilu** [reply]
I got some idea on MCUboot. I am able to build code for b_u585i_iot02a. However same does not work for nucleo_u575zi_q board.
I found one ticket on MCUboot issue
[https://github.com/zephyrproject-rtos/zephyr/issues/47781](https://github.com/zephyrproject-rtos/zephyr/issues/47781)

**Comment by VicLi0621** [comment]
Hi Sir,

I want to use the STM32U5A9 which is one member of the STM32U5 series and follow your suggestion to check the code below the link. It looks like Zephyr still doesn't support STM32U5A9. Is it right? If the answer is yes, it looks like I need to port the STM32U5A9 to Zephyr on SoC level(STM32U5A9). I can find the board porting guideline from Zephyr documentation but don't find the SoC porting guideline. Please kindly let us know if you have any suggestions. Thanks.

Board support - Github
https://github.com/zephyrproject-rtos/zephyr/tree/main/dts/arm/st/u5

board porting guideline
https://docs.zephyrproject.org/3.2.0/hardware/porting/board_porting.html

**Reply by erwango** [reply]
You are correct, `STM32U5A9` is not yet ported in Zephyr.
It would be a welcome addition to add it. There is no SoC porting Guideline.
What I suggest to do is to look at following folders and add the new SoC by following what is done for a similar SoC:
https://github.com/zephyrproject-rtos/zephyr/tree/main/dts/arm/st/u5
https://github.com/zephyrproject-rtos/zephyr/tree/main/soc/arm/st_stm32/stm32u5/
